biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotations.java

```diff
@@ -26,55 +50,150 @@ public boolean analyzeJar(Analyzer analyzer) throws Exception {
+
+							for (int i = 0; i < definition.service.length; i++) {
+								Descriptors.TypeRef tr = definition.service[i];
+								objectClass[i] = tr.getFQN();
+							}
+							Arrays.sort(objectClass);
+							addProvidesHeader(objectClass, provides);
+						}
+						for (ReferenceDef ref : definition.references.values()) {
+							String objectClass = ref.service;
+							addRequireHeader(objectClass, requires);
+						}
+						maxVersion = definition.max(maxVersion, definition.version);
 					}
 				}
 			}
 		}
+		if (flags.contains(Flag.extender)
+				|| (packageVersion != null && packageVersion.compareTo(AnnotationReader.V1_3) >= 0)
+				|| maxVersion.compareTo(AnnotationReader.V1_3) >= 0) {
+			addExtenRequireHeader(requires);
+		}
 		sc = Processor.append(names.toArray(new String[names.size()]));
 		analyzer.setProperty(Constants.SERVICE_COMPONENT, sc);
+		updateHeader(analyzer, Constants.REQUIRE_CAPABILITY, requires);
+		updateHeader(analyzer, Constants.PROVIDE_CAPABILITY, provides);
 		return false;
 	}
 
 	/**
 	 * Verify that the component definition has a version that is <= than the
 	 * version of the component package that we build against.
+	 * 
+	 * @param version
+	 *            package version of org.osgi.service.component exported in
+	 *            build environment.
 	 */
-	private void verifyVersion(Analyzer analyzer, ComponentDef definition) throws Exception {
-		PackageRef component = analyzer.getPackageRef("org.osgi.service.component");
+	private void verifyVersion(Analyzer analyzer, ComponentDef definition, Version v) throws Exception {
+		if (v != null) {
+			if (definition.version.compareTo(v) > 0) {
+				SetLocation error = analyzer
+						.error("Generating XML for %s in type %s that uses a namespace version %s while you are building against %s",
+								definition.name, definition.implementation, definition.version, v);
+
+				error.details(analyzer.getPackageRef("org/osgi/service/component"));
+				analyzer.setTypeLocation(error, definition.implementation);
+			}
+		}
+	}
+
+	private Version getPackageVersion(Analyzer analyzer) {
+		PackageRef component = analyzer.getPackageRef("org/osgi/service/component");
 		Attrs attrs = analyzer.getClasspathExports().get(component);
 		if (attrs != null) {
 			String version = attrs.getVersion();
 			if (version != null && Verifier.isVersion(version)) {
-				Version v = new Version(version);
-				if (definition.version.compareTo(v) > 0) {
-					SetLocation error = analyzer
-							.error("Generating XML for %s in type %s that uses a namespace version %s while you are building against %s",
-									definition.name, definition.implementation, definition.version, v);
-
-					error.details(component);
-					analyzer.setTypeLocation(error, definition.implementation);
-				}
+				return new Version(version);
+			}
+		}
+		return null;
+	}
+
+	private void addProvidesHeader(String[] objectClass, Set<String> provides) {
```

Comment:
I think this should be called addProvideHeader (not Provides) since the require method is addRequireHeader.